packages/node-resolve/src/resolveImportSpecifiers.js

```diff
@@ -0,0 +1,197 @@
+import fs from 'fs';
+import path from 'path';
+import { promisify } from 'util';
+
+import resolve from 'resolve';
+
+import { getPackageName } from './util';
+import { exists, realpath } from './fs';
+
+const resolveImportPath = promisify(resolve);
+const readFile = promisify(fs.readFile);
+
+const pathNotFoundError = (subPath, pkgPath) =>
+  new Error(`Package subpath '${subPath}' is not defined by "exports" in ${pkgPath}`);
+
+function findExportKeyMatch(exportMap, subPath) {
+  for (const [key, value] of Object.entries(exportMap)) {
+    if (key.endsWith('*') && value.includes('*')) {
+      // star match: "./foo/*": "./foo/*.js"
+      const keyWithoutStar = key.substring(0, key.length - 1);
+      if (subPath.startsWith(keyWithoutStar)) {
+        return key;
+      }
```

Comment:
There isn't any requirement that the subpath be the same - it is fine to do `"exports": { "./virtual/*": "./local/*/index.js" }`.

Reply:
This checks whether the user's import starts with the part that comes before the star in the key. For example:

```json
"exports": { "./virtual/*": "./local/*/index.js" }
```

should match:

`import './virtual/foo`

but not:

`import './foo`

Correct?

Reply:
Ah right, seems good!